src/main/java/com/gdsc/boilerplate/springboot/security/annotation/AuthenticatedUser.java

```diff
@@ -0,0 +1,11 @@
+package com.gdsc.boilerplate.springboot.security.annotation;
+
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+
+import java.lang.annotation.*;
+
+@Target({ElementType.PARAMETER, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@AuthenticationPrincipal
+public @interface AuthenticatedUser {}
```

Comment:
why not use AuthenticationPrincipal anotation directly, i can't understand. Can you explain for me ?

Reply:
đã update